docs/en/multi_modal/internvl.md

```diff
@@ -1,3 +1,154 @@
-# InternVL
+# InternVL2
 
-TODO
+## Introduction
+
+InternVL is an open source vision-language base model that expands the Vision Transformer (ViT) to 600 million parameters and aligns with the Large Language Model (LLM). It is the largest open-source vision/vision-language foundation model (14B) to date, achieving 32 state-of-the-art performance on a wide range of tasks such as visual perception, cross-modal retrieval, multimodal dialogue, etc. LMDeploy supports InternVL series of models. The following uses InternVL2-8B as an example to demonstrate its usage.
+
+## Quick Start
+
+### Installation
+
+Please install LMDeploy by following the [installation guide](../installation.md), and install other packages that InternVL2 needs
+
+```shell
+pip install timm
+```
+
+### Offline inference pipeline
+
+The following sample code shows the basic usage of VLM pipeline. For more examples, please refer to [VLM Offline Inference Pipeline](./vl_pipeline.md)
+
+```python
+from lmdeploy import pipeline
+from lmdeploy.vl import load_image
+
+pipe = pipeline('OpenGVLab/InternVL2-8B')
+
+image = load_image('https://raw.githubusercontent.com/open-mmlab/mmdeploy/main/tests/data/tiger.jpeg')
+response = pipe((f'describe this image', image))
+print(response)
+```
+
+## More examples
+
+<details>
+  <summary>
+    <b>multi-image multi-round conversation, combined images</b>
+  </summary>
+
+```python
+from lmdeploy import pipeline
+from lmdeploy.vl import load_image
+from lmdeploy.vl.utils import encode_image_base64
+from lmdeploy.vl.constants import IMAGE_TOKEN
+
+pipe = pipeline('OpenGVLab/InternVL2-8B', log_level='INFO')
+messages = [
+    dict(role='user', content=[
+        dict(type='text', text=f'<img>{IMAGE_TOKEN}{IMAGE_TOKEN}</img>\nDescribe the two images in detail.'),
+        dict(type='image_url', image_url=dict(max_dynamic_patch=12, url='https://raw.githubusercontent.com/OpenGVLab/InternVL/main/internvl_chat/examples/image1.jpg')),
+        dict(type='image_url', image_url=dict(max_dynamic_patch=12, url='https://raw.githubusercontent.com/OpenGVLab/InternVL/main/internvl_chat/examples/image2.jpg'))
+    ])
+]
+out = pipe(messages, top_k=1)
```

Comment:
better put `top_k` in `GenerationConfig`. Directly input `top_k` may not be valid.